lib/options.go

```diff
@@ -365,3 +366,30 @@ func (o Options) Apply(opts Options) Options {
 	}
 	return o
 }
+
+// GetPrettyJSON is a massive hack that works around the fact that some
+// of the null-able types used in Options are marshalled to `null` when
+// their `valid` flag is false.
+// TODO: figure out something better or at least use reflect to do it, that
```

Comment:
In the future we could perhaps compare the keys from the user supplied JSON file + any options from the converter itself (like `maxRedirects`) with the keys in the options struct and only marshal the ones that have been specified (via some sort of intermediary data struct/container, like you use below for removing the null values). Maybe that's what you mean by "use reflect to do it".

Reply:
Not exactly, we can just enumerate struct fields with `reflect` (in order, compared to the unordered map traversal in Go) and check each one for a nil or a `false` value of its `Valid` flag. That way we'll also be able to output invalid JSON, but valid JS object syntax like this (notice the lack of quotes around the key name):
```
{
    maxRedirects: 0
};
```

Reply:
I didn't do it now because reflect is quite finicky and I couldn't be sure that I'd get all of the edge cases correctly..

Reply:
👍